### ISSUE TITLE
tatic instances could also be transformed with `prefer_const_constructors`

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#5049]
**Describe the issue**
A clear and concise description of what the issue is.

If you're seeing unexpected behavior in a lint rule, please include the name of the rule in the issue description.

**To Reproduce**
Ideally, please include an example.

**Expected behavior**

**Additional context**

**Comment by bwilkerson** [comment]
I suspect that this would want to be a separate lint as it would probably want to flag any static final field whose initializer is a constant expression.

I included 'final' in the description above because changing a variable (especially a public variable) from `static` to `static const` is a significant change. If the field is already `final` then changing it to `const` is mostly just an optimization.

**Comment by FMorschel** [comment]
> I included 'final' in the description above because changing a variable (especially a public variable) from static to static const is a significant change.

So if the static variable is private or if the class itself is private the lint could show. I agree. 

I'm not so good in naming things and I can't seem to find your issue on naming lints (for wording and stuff - saw it once when browsing here) so I'll let others suggest a name for it.

**Comment by FMorschel** [comment]
## prefer_const_unmodified_static_variables

## Description

Should use `const` for static variables that are declared as final but are initialized with a constant value.

## Details

Promoting the practice of marking static variables as `const` instead of `final` when they are initialized with a constant value. Using const ensures that the variables are compile-time constants, which can improve performance and code clarity by avoiding unnecessary allocations and guaranteeing immutability.

## Kind

This rule enforces style advice and promotes code optimization by encouraging the use of `const` for static variables that are unmodified.

## Bad Examples

```dart
static final myList = const [1, 2, 3];
static final myMap = const {'key': 'value'};
```

## Good Examples 

```dart
static const myList = [1, 2, 3];
static const myMap = {'key': 'value'};
```

## Discussion

This proposal addresses a gap in the current Dart lints by targeting static final variables that could be declared as `const`. By ensuring that these variables are declared as `const` where applicable, the lint helps prevent missed opportunities for optimization and promotes clearer, more maintainable code.

I believe this should be added to Effective Dart. Since it is an extension of `prefer_const_declarations`.

### Discussion checklist

- [x] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [ ] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [ ] If there's any prior art (e.g., in other linters), please add references here.
- [x] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [x] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

**Comment by bwilkerson** [comment]
@munificent (For the suggestion of adding this to Effective Dart.)

**Comment by munificent** [comment]
> If the field is already `final` then changing it to `const` is mostly just an optimization.

*Mostly*, but it does mean that the author of that API is committing to that static field being usable in constant expressions and it becomes a breaking API change to remove the `const`.
